File: 5.implementation/VT/redsun/src/main/webapp/static/js/services/TrendsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('trendService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(trend) {
			var urlServer = $rootScope.makePostURL("/trend/insert");
			return $http.post(urlServer, trend);
		}
		
		// Update.
		, update: function(trendId, trend) {
			var urlServer = $rootScope.makePostURL("/trend/update/" + trendId);
			return $http.put(urlServer, trend);
		}
		
		// Delete.
		, delete: function(trendId) {
			var urlServer = $rootScope.makePostURL("/trend/delete/" + trendId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(trendId) {
			var url = $rootScope.makeGetURL("/trend/getbyid/" + trendId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, trend) {
			var url = $rootScope.makeGetURL("/trend/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: trend });
		}
		
	}// return.
});

File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/TrendsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Trends;
import com.redsun.service.entities.Result;

/**
 * Trends Service interface
 */
public interface TrendsService {
	
    // Insert.
	Result insert(final Trends trends) ;

    // Update.
	Result update(final Trends trends) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listTrendsForPageAndFilter(final int itemsPerPage, final int pageNo, final Trends trends);

}
